compiler/rustc_middle/src/ty/adt.rs

```diff
@@ -89,62 +90,62 @@ bitflags! {
 /// where `x` here represents the `DefId` of `S.x`. Then, the `DefId`
 /// can be used with [`TyCtxt::type_of()`] to get the type of the field.
 #[derive(TyEncodable, TyDecodable)]
-pub struct AdtDef {
+pub struct AdtDefS {
```

Comment:
Can the naming convention be changed to something like `AdtDefData`? (Like e.g. `Span` and `SpanData`.)

I know that the `S` comes from `TyS`, but `TyS` doesn't look like an intentional choice, but rather like some horrible legacy naming that nobody likes but has no time to rename.

Reply:
I have updated the code to use `AdtData`.